src/GraphBLAS-sharp.Backend/Vector/COOVector/COOVector.fs

```diff
@@ -0,0 +1,542 @@
+                        isLeftBitMap.[i] <- 1 @>
+
+        let kernel = clContext.Compile(merge)
+
+        fun (processor: MailboxProcessor<_>) (firstIndices: ClArray<int>) (firstValues: ClArray<'a>) (secondIndices: ClArray<int>) (secondValues: ClArray<'b>) ->
+
+            let firstSide = firstIndices.Length
+
+            let secondSide = secondIndices.Length
+
+            let sumOfSides =
+                firstIndices.Length + secondIndices.Length
+
+            let allIndices =
+                clContext.CreateClArray<int>(
+                    sumOfSides,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    deviceAccessMode = DeviceAccessMode.ReadWrite,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let firstResultValues =
+                clContext.CreateClArray<'a>(
+                    sumOfSides,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    deviceAccessMode = DeviceAccessMode.ReadWrite,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let secondResultValues =
+                clContext.CreateClArray<'b>(
+                    sumOfSides,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    deviceAccessMode = DeviceAccessMode.ReadWrite,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let isLeftBitmap =
+                clContext.CreateClArray<int>(
+                    sumOfSides,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    deviceAccessMode = DeviceAccessMode.ReadWrite,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let ndRange =
+                Range1D.CreateValid(sumOfSides, workGroupSize)
+
+            let kernel = kernel.GetKernel()
+
+            processor.Post(
+                Msg.MsgSetArguments
+                    (fun () ->
+                        kernel.KernelFunc
+                            ndRange
+                            firstSide
+                            secondSide
+                            sumOfSides
+                            firstIndices
+                            firstValues
+                            secondIndices
+                            secondValues
+                            allIndices
+                            firstResultValues
+                            secondResultValues
+                            isLeftBitmap)
+            )
+
+            processor.Post(Msg.CreateRunMsg<_, _>(kernel))
+
+            allIndices, firstResultValues, secondResultValues, isLeftBitmap
+
+    let private preparePositionsAtLeasOne<'a, 'b, 'c when 'a: struct and 'b: struct and 'c: struct>
+        (clContext: ClContext)
+        (opAdd: Expr<AtLeastOne<'a, 'b> -> 'c option>)
+        (workGroupSize: int)
+        =
+
+        let preparePositions =
```

Comment:
Looks similar to [this function](https://github.com/YaccConstructor/GraphBLAS-sharp/blob/net5/src/GraphBLAS-sharp.Backend/Matrix/COOMatrix/COOMatrix.fs#L104).